src/org/thoughtcrime/securesms/ConversationActivity.java

```diff
@@ -158,11 +158,11 @@ public class ConversationActivity extends PassphraseRequiredActionBarActivity
 {
   private static final String TAG = ConversationActivity.class.getSimpleName();
 
-  public static final String ACCOUNT_ID_EXTRA        = TAG + ".account_id";
-  public static final String CHAT_ID_EXTRA           = TAG + ".chat_id";
-  public static final String FROM_ARCHIVED_CHATS_EXTRA = TAG + ".from_archived";
-  public static final String TEXT_EXTRA              = TAG + ".draft_text";
-  public static final String STARTING_POSITION_EXTRA = TAG + ".starting_position";
+  public static final String ACCOUNT_ID_EXTRA        = "account_id";
```

Comment:
why the solution is to make all EXTRA be the same in all classes instead of using the appropriate EXTRA of the target activity when creating an intent? by reading this PR I don't really see the place that were causing the issue (setting a wrong extra) this is not a good signal IMHO

Reply:
Because I just reverted the commit that caused the bug.

The really correct solution would IMO be to have one definition of the variable in some central place, and then import it from everywhere. A wrong import shouldn't be able to cause a bug.

The cause of the bug is that `ShareActivity` is:
```java
      composeIntent = getBaseShareIntent(ConversationActivity.class);
      composeIntent.putExtra(EXTRA_CHAT_ID, chatId);
      RelayUtil.setSharedUris(composeIntent, resolvedExtras);
      startActivity(composeIntent);
```

which doesn't use ConversationActivity's "chat_id", but ShareActivity's.

Reply:
> which doesn't use ConversationActivity's "chat_id", but ShareActivity's.

that is the real bug then

Reply:
> A wrong import shouldn't be able to cause a bug.

putting all extras in a single class would do as well, but I think intent creator must be careful anyway to use the extras of the same class you are launching, it doesn't makes much sense that you use the extras of other classes

Reply:
If we only fix this single bug, then I'd be afraid that https://github.com/deltachat/deltachat-android/commit/8962476a21c67fd943a496f759422a4b2e6cd3e3 introduced even more bugs which we'll then also release.

> I think intent creator must be careful anyway to use the extras of the same class you are launching

Well, obviously someone was not careful when using the wrong variable here, whenever possible one shouldn't have to rely on others or oneself being careful (I mean, that's what "software quality" usually boils down to, with writing tests, avoiding code duplication, ...)

---

I'm didn't get which solutions would be OK for you - would you go with just merging this PR if it fixes the bug? This PR plus using the correct variable there (I tried to keep the changes minimal and go back to a known-good state right before a release, but changing the variable here wouldn't be a problem)? This PR plus using the same variable from everywhere?

Reply:
yes, this PR is fine, but I think also fixing the original issue and use the right variable should be done in this PR as well